src/cc65/declare.c

```diff
@@ -2079,8 +2089,14 @@ static unsigned ParseStructInit (Type* T, int AllowFlexibleMembers)
 
             /* Account for the data and output it if we have a full word */
             SI.ValBits += Entry->V.B.BitWidth;
-            CHECK (SI.ValBits <= INT_BITS);
-            if (SI.ValBits == INT_BITS) {
+            /* Make sure unsigned is big enough to hold the value, 30 bits.
+            ** This is 30 and not 32 bits because a 16-bit bit-field will
+            ** always be byte aligned, so will have padding before it.
+            ** 15 bits twice is the most we can have.
+            */
+            CHECK (SI.ValBits <= CHAR_BIT * sizeof(SI.BitVal));
```

Comment:
Let me know if assuming 32-bit `unsigned`s on the host platform is a problem.

Reply:
Does that mean at least 32-bit unsigned (in contrast to 16-bit unsigned)?

Reply:
Yes, currently the host platform needs 32-bit `unsigned`. If that's not ok, the next easiest thing would be to use `uint32_fast_t` or `unsigned long` and assume that's 32-bits.